manifests/install.pp

```diff
@@ -23,9 +22,21 @@
   }
 
   if String($globus::version) == '5' {
-    package { $globus::package_name:
-      ensure => 'present',
+    case $facts['os']['release']['major'] {
+      '8':  {
+        exec { 'dnf reset':
+          command => '/bin/dnf module -y disable mod_auth_openidc',
+          unless  => '/bin/rpm -aq | grep $globus::package_name',
+        }
+        package { $globus::package_name:
+          ensure => 'present',
+        }
+      }
+      default: {
+        package { $globus::package_name:
+          ensure => 'present',
+        }
```

Comment:
Because the package does not change between conditions, I'd recommend moving it out of the case statement.